src/execution/index/art/art.cpp

```diff
@@ -811,52 +845,61 @@ void ART::VerifyExistence(DataChunk &chunk, VerifyExistenceType verify_type, str
 		if (keys[i].Empty()) {
 			continue;
 		}
+
 		Node *node_ptr = Lookup(tree, keys[i], 0);
-		bool throw_exception =
-		    verify_type == VerifyExistenceType::APPEND_FK ? node_ptr == nullptr : node_ptr != nullptr;
-		if (!throw_exception) {
+
+		// APPEND_FK, i.e., the key must exist in the PK/UNIQUE table
```

Comment:
Is there any chance you could move the adjustments made here to another PR?
I have had to alter this code quite a bit as part of UPSERT so we're bound to get a bunch of merge conflicts there

Reply:
I'd like to keep them in for now, but you can just overwrite them when merging, no?

Reply:
Yea fair enough, on second inspection they look pretty minor, I think I just read VerifyExistence and got PTSD, because like 2/3rds of the time spent on UPSERT was spent untangling this code